components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -367,8 +368,11 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
                 if (log.isDebugEnabled()) {
                     log.debug(e.getMessage(), e);
                 }
-                FrameworkUtils.sendToRetryPage(request, response, context, "suspicious.authentication.attempts",
-                        "suspicious.authentication.attempts.description");
+                log.warn("Session nonce cookie validation has failed. Hence, restarting the login flow.");
+                context = (AuthenticationContext) context.getProperty(FrameworkConstants.INITIAL_CONTEXT);
+                context.setProperty(FrameworkConstants.RESTART_LOGIN_FLOW, true);
+                FrameworkUtils.addAuthenticationContextToCache(context.getContextIdentifier(), context);
+                handle(request, response);
```

Comment:
Here we are adding a new entry to the Session table which will be redundant. 

can we add this `RESTART_LOGIN_FLOW` to the request and call handle method? Inside the handle method based on this request parameter, we can get back the initial context from the context and continue the flow.

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/4509/commits/b913f3378040a4c77e6f2376117ae1fa7f7f9009